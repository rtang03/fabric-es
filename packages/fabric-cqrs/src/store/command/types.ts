import { Wallet } from 'fabric-network';
import { BaseEvent } from '../../types';

export interface CreateAction {
  type: string;
  payload?: {
    tx_id: string;
    args: {
      entityName: string;
      id: string;
      version: number;
      events: BaseEvent[];
      isPrivateData: boolean;
    };
    enrollmentId?: string;
    channelName?: string;
    connectionProfile?: string;
    wallet?: Wallet;
  };
}

export interface DeleteByEntityIdAction {
  type: string;
  payload?: {
    tx_id: string;
    args: {
      entityName: string;
      id: string;
      isPrivateData: boolean;
    };
  };
}

export interface DeleteByEntityIdCommitIdAction {
  type: string;
  payload?: {
    tx_id: string;
    args: {
      entityName: string;
      id: string;
      commitId: string;
      isPrivateData: boolean;
    };
  };
}

export interface QueryByEntityIdAction {
  type: string;
  payload?: {
    tx_id: string;
    args: {
      entityName: string;
      id: string;
      isPrivateData: boolean;
    };
  };
}

export interface QueryByEntityNameAction {
  type: string;
  payload?: {
    tx_id: string;
    args: {
      entityName: string;
      isPrivateData: boolean;
    };
  };
}

export interface QueryByEntIdCommitIdAction {
  type: string;
  payload?: {
    tx_id: string;
    args: {
      entityName: string;
      id: string;
      commitId: string;
      isPrivateData: boolean;
    };
  };
}
