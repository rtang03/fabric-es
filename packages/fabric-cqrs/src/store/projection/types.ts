import { Commit } from '../../types';

export interface MergeEntityAction {
  type: string;
  payload?: {
    tx_id: string;
    args: {
      commit: Commit;
    };
  };
}

export interface MergeEntityBatchAction {
  type: string;
  payload?: {
    tx_id: string;
    args: {
      entityName: string;
      commits: Record<string, Commit>;
    };
  };
}
