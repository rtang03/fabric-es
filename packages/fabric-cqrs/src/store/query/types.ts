import type { FTSearchParameters } from 'redis-modules-sdk';
import type { Commit } from '../../types';

export interface DeleteCommitByEntityIdAction {
  type: string;
  payload?: {
    tx_id: string;
    args: {
      entityName: string;
      id: string;
    };
  };
}

export interface DeleteCommitByEntityNameAction {
  type: string;
  payload?: {
    tx_id: string;
    args: { entityName: string };
  };
}

export interface QueryByEntityNameAction {
  type: string;
  payload?: {
    tx_id: string;
    args: {
      entityName: string;
    };
  };
}

export interface QueryByEntityIdAction {
  type: string;
  payload?: {
    tx_id: string;
    args: {
      entityName: string;
      id: string;
    };
  };
}

export interface MergeAction {
  type: string;
  payload?: {
    tx_id: string;
    args: {
      commit: Commit;
    };
  };
}

export interface MergeBatchAction {
  type: string;
  payload?: {
    tx_id: string;
    args: {
      entityName: string;
      commits: Record<string, Commit>;
    };
  };
}

export interface CIdxSearchAction {
  type: string;
  payload?: {
    tx_id: string;
    args: {
      query: string;
      param?: FTSearchParameters;
      countTotalOnly?: boolean;
    };
  };
}

export interface EIdxSearchAction {
  type: string;
  payload?: {
    tx_id: string;
    args: {
      entityName: string;
      query: string;
      param?: FTSearchParameters;
      countTotalOnly?: boolean;
    };
  };
}

export interface FindAction {
  type: string;
  payload?: {
    tx_id: string;
    args: {
      entityName: string;
      byId?: string;
      byDesc?: string;
      where?: any;
    };
  };
}

export interface NotifyAction {
  type: string;
  payload?: {
    tx_id: string;
    args: {
      creator: string;
      commitId: string;
      entityName: string;
      id: string;
    };
  };
}
