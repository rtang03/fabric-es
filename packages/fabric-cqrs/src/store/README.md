### Note:

`store` is the standard Redux architecture. Along with `redux-observables`, it acts a Reactive Functional Programming
layer between CommandHandler / QueryHandler, and data layer ( Redis and Hyperledger ). It is the core CQRS pattern.

- Projection epics: when new entity arrives the Hyperledger peer, the contract listener will send it to Redis
- Command epics: write to Hyperledger
- Query epics: read from Redis
- Reconcile: reconcile commit, during bootstrapping steps

The codes here are all internal implementation, and will not be published via typedoc.
