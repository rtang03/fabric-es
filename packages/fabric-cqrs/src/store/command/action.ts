/**
 * @packageDocumentation
 * @hidden
 */
import { getAction, getErrorAction, getSuccessAction } from '../utils';
import type {
  CreateAction,
  DeleteByEntityIdAction,
  DeleteByEntityIdCommitIdAction,
  QueryByEntIdCommitIdAction,
  QueryByEntityIdAction,
  QueryByEntityNameAction,
  TrackAction,
} from './types';

const CREATE = '[Command] Create';
const CREATE_ERROR = '[Command] Create Error';
const CREATE_SUCCESS = '[Command] Create Success';
const DELETE_BY_ENTITYID = '[Command] Delete commit By entityName and id';
const DELETE_ERROR = '[Command] Delete Error';
const DELETE_SUCCESS = '[Command] Delete Success';
const DELETE_BY_ENTITYID_COMMITID = '[Command] Delete By entityId and commitId';
const QUERY_BY_ENTITY_ID = '[Command] Query commit by entity id';
const QUERY_BY_ENTITY_NAME = '[Command] Query commit by entityName';
const QUERY_BY_ENTITYID_COMMITID = '[Command] Query by entity id and commitId';
const QUERY_ERROR = '[Command] Query Error';
const QUERY_SUCCESS = '[Command] Query Success';
const TRACK = '[Entity-C] Track';
const TRACK_ERROR = '[Entity-C] Track Error';
const TRACK_SUCCESS = '[Entity-C] Track Success';

export const action = {
  CREATE,
  CREATE_ERROR,
  CREATE_SUCCESS,
  DELETE_BY_ENTITYID,
  DELETE_ERROR,
  DELETE_SUCCESS,
  DELETE_BY_ENTITYID_COMMITID,
  QUERY_BY_ENTITY_ID,
  QUERY_BY_ENTITY_NAME,
  QUERY_BY_ENTITYID_COMMITID,
  QUERY_ERROR,
  QUERY_SUCCESS,
  TRACK,
  TRACK_ERROR,
  TRACK_SUCCESS,
  create: getAction<CreateAction>(CREATE),
  deleteCommitByEntityId: getAction<DeleteByEntityIdAction>(DELETE_BY_ENTITYID),
  deleteByEntityIdCommitId: getAction<DeleteByEntityIdCommitIdAction>(DELETE_BY_ENTITYID_COMMITID),
  queryByEntityId: getAction<QueryByEntityIdAction>(QUERY_BY_ENTITY_ID),
  queryByEntityName: getAction<QueryByEntityNameAction>(QUERY_BY_ENTITY_NAME),
  queryByEntIdCommitId: getAction<QueryByEntIdCommitIdAction>(QUERY_BY_ENTITYID_COMMITID),
  track: getAction<TrackAction>(TRACK),
  createError: getErrorAction(CREATE_ERROR),
  createSuccess: getSuccessAction(CREATE_SUCCESS),
  deleteError: getErrorAction(DELETE_ERROR),
  deleteSuccess: getSuccessAction(DELETE_SUCCESS),
  queryError: getErrorAction(QUERY_ERROR),
  querySuccess: getSuccessAction(QUERY_SUCCESS),
  trackError: getErrorAction(TRACK_ERROR),
  trackSuccess: getSuccessAction(TRACK_SUCCESS),
};
