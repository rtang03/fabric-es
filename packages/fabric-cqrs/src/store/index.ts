/**
 * @packageDocumentation
 * @hidden
 */
export * from './store';
