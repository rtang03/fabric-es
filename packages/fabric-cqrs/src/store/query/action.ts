import { getAction, getErrorAction, getSuccessAction } from '../utils';
import type {
  DeleteCommitByEntityIdAction,
  DeleteCommitByEntityNameAction,
  MergeAction,
  MergeBatchAction,
  QueryByEntityIdAction,
  QueryByEntityNameAction,
  EIdxSearchAction,
  CIdxSearchAction,
  FindAction,
  NotifyAction,
  GetNotificationsAction,
  GetNotificationAction,
  ClearNotificationAction,
  ClearNotificationsAction,
  DeleteEntityByEntityNameAction,
} from './types';

const DELETE_COMMIT_BY_ENTITYNAME = '[Query-Db] Delete commits by EntityName';
const DELETE_COMMIT_BY_ENTITY_ID = '[Query-Db] Delete commits by entityId';
const DELETE_ERROR = '[Query-Db] Delete Error';
const DELETE_SUCCESS = '[Query-Db] Delete Success';
const QUERY_BY_ENTITYNAME = '[Query-Db] Query commits By EntityName';
const QUERY_BY_ENTITY_ID = '[Query-Db] Query commit By entityId';
const QUERY_ERROR = '[Query-Db] Query Error';
const QUERY_SUCCESS = '[Query-Db] Query Success';
const MERGE_COMMIT = '[Query-Db] Merge commit';
const MERGE_COMMIT_ERROR = '[Query-Db] Merge Error';
const MERGE_COMMIT_SUCCESS = '[Query-Db] Merge Success';
const MERGE_COMMIT_BATCH = '[Query-Db] Merge records of entities';
const MERGE_COMMIT_BATCH_ERROR = '[Query-Db] Merge Batch Error';
const MERGE_COMMIT_BATCH_SUCCESS = '[Query-Db] Merge Batch Success';
const CIDX_SEARCH = '[Query-Db] cidx Search';
const EIDX_SEARCH = '[Query-Db] eidx Search';
const SEARCH_SUCCESS = '[Query-Db] Search Success';
const SEARCH_ERROR = '[Query-Db] Search Error';
const FIND = '[Query-Db] Find';
const FIND_SUCCESS = '[Query-Db] Find Success';
const FIND_ERROR = '[Query-Db] Find Error';
const NOTIFY = '[Query-Db] Notify';
const NOTIFY_SUCCESS = '[Query-Db] Notify Success';
const NOTIFY_ERROR = '[Query-Db] Notify Error';
const GET_NOTIFICATION = '[Query-Db] Get notification';
const GET_NOTIFICATIONS = '[Query-Db] Get notifications';
const GET_NOTI_SUCCESS = '[Query-Db] Get Notification / Notifications Success';
const GET_NOTI_ERROR = '[Query-Db] Get Notification / Notifications Error';
const CLEAR_NOTIFICATION = '[Query-Db] Clear notification';
const CLEAR_NOTIFICATIONS = '[Query-Db] Clear notifications';
const CLEAR_NOTI_SUCCESS = '[Query-Db] Clear Notification / Notifications Success';
const CLEAR_NOTI_ERROR = '[Query-Db] Clear Notification / Notifications Error';
const DELETE_ENTITY_BY_ENTITYNAME = '[Query-Db] Delete entity by EntityName';
const DELETE_ENTITY_SUCCESS = '[Query-Db] Delete entity by EntityName Success';
const DELETE_ENTITY_ERROR = '[Query-Db] Delete entity by EntityName Error';

export const action = {
  DELETE_COMMIT_BY_ENTITYNAME,
  DELETE_COMMIT_BY_ENTITY_ID,
  DELETE_ERROR,
  DELETE_SUCCESS,
  QUERY_BY_ENTITYNAME,
  QUERY_BY_ENTITY_ID,
  QUERY_ERROR,
  QUERY_SUCCESS,
  MERGE_COMMIT,
  MERGE_COMMIT_ERROR,
  MERGE_COMMIT_SUCCESS,
  MERGE_COMMIT_BATCH,
  MERGE_COMMIT_BATCH_ERROR,
  MERGE_COMMIT_BATCH_SUCCESS,
  CIDX_SEARCH,
  EIDX_SEARCH,
  SEARCH_SUCCESS,
  SEARCH_ERROR,
  FIND,
  FIND_SUCCESS,
  FIND_ERROR,
  NOTIFY,
  NOTIFY_SUCCESS,
  NOTIFY_ERROR,
  GET_NOTIFICATION,
  GET_NOTIFICATIONS,
  GET_NOTI_SUCCESS,
  GET_NOTI_ERROR,
  CLEAR_NOTIFICATION,
  CLEAR_NOTIFICATIONS,
  CLEAR_NOTI_SUCCESS,
  CLEAR_NOTI_ERROR,
  DELETE_ENTITY_BY_ENTITYNAME,
  DELETE_ENTITY_SUCCESS,
  DELETE_ENTITY_ERROR,
  deleteCommitByEntityId: getAction<DeleteCommitByEntityIdAction>(DELETE_COMMIT_BY_ENTITY_ID),
  deleteCommitByEntityName: getAction<DeleteCommitByEntityNameAction>(DELETE_COMMIT_BY_ENTITYNAME),
  mergeCommit: getAction<MergeAction>(MERGE_COMMIT),
  mergeCommitBatch: getAction<MergeBatchAction>(MERGE_COMMIT_BATCH),
  queryByEntityName: getAction<QueryByEntityNameAction>(QUERY_BY_ENTITYNAME),
  queryByEntityId: getAction<QueryByEntityIdAction>(QUERY_BY_ENTITY_ID),
  deleteError: getErrorAction(DELETE_ERROR),
  deleteSuccess: getSuccessAction(DELETE_SUCCESS),
  queryError: getErrorAction(QUERY_ERROR),
  querySuccess: getSuccessAction(QUERY_SUCCESS),
  mergeCommitError: getErrorAction(MERGE_COMMIT_ERROR),
  mergeCommitSuccess: getSuccessAction(MERGE_COMMIT_SUCCESS),
  mergeCommitBatchError: getErrorAction(MERGE_COMMIT_BATCH_ERROR),
  mergeCommitBatchSuccess: getSuccessAction(MERGE_COMMIT_BATCH_SUCCESS),
  cIdxSearch: getAction<CIdxSearchAction>(CIDX_SEARCH),
  eIdxSearch: getAction<EIdxSearchAction>(EIDX_SEARCH),
  searchSuccess: getSuccessAction(SEARCH_SUCCESS),
  searchError: getErrorAction(SEARCH_ERROR),
  find: getAction<FindAction>(FIND),
  findSuccess: getSuccessAction(FIND_SUCCESS),
  findError: getErrorAction(FIND_ERROR),
  notify: getAction<NotifyAction>(NOTIFY),
  notifySuccess: getSuccessAction(NOTIFY_SUCCESS),
  notifyError: getErrorAction(NOTIFY_ERROR),
  getNotification: getAction<GetNotificationAction>(GET_NOTIFICATION),
  getNotifications: getAction<GetNotificationsAction>(GET_NOTIFICATIONS),
  getNotiSuccess: getSuccessAction(GET_NOTI_SUCCESS),
  getNotiError: getErrorAction(GET_NOTI_ERROR),
  clearNotification: getAction<ClearNotificationAction>(CLEAR_NOTIFICATION),
  clearNotifications: getAction<ClearNotificationsAction>(CLEAR_NOTIFICATIONS),
  clearNotiSuccess: getSuccessAction(CLEAR_NOTI_SUCCESS),
  clearNotiError: getErrorAction(CLEAR_NOTI_ERROR),
  deleteEntityByEntityName: getAction<DeleteEntityByEntityNameAction>(DELETE_ENTITY_BY_ENTITYNAME),
  deleteEntityByEntityNameSuccess: getSuccessAction(DELETE_ENTITY_SUCCESS),
  deleteEntityByEntityNameError: getErrorAction(DELETE_ENTITY_ERROR),
};
