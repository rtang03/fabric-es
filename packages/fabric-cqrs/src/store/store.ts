/**
 * @packageDocumentation
 * @hidden
 */
import { Gateway, Network } from 'fabric-network';
import { applyMiddleware, combineReducers, createStore, Store } from 'redux';
import { combineEpics, createEpicMiddleware } from 'redux-observable';
import type { Logger } from 'winston';
import type { Reducer } from '../types';
import type { QueryDatabase } from '../types';
import { commandEpic, reducer as write } from './command';
import { projectionEpic, reducer as projection } from './projection';
import { queryEpic, reducer as query } from './query';
import { reconcileEpic, reducer as reconcile } from './reconcile';

const rootEpic = combineEpics(...commandEpic, ...projectionEpic, ...queryEpic, ...reconcileEpic);

const rootReducer = combineReducers({ write, projection, query, reconcile });

export const getStore: (options: {
  queryDatabase: QueryDatabase;
  gateway: Gateway;
  network: Network;
  reducers: Record<string, Reducer>;
  logger: Logger;
}) => Store = ({ queryDatabase, gateway, network, reducers, logger }) => {
  const epicMiddleware = createEpicMiddleware({
    dependencies: {
      queryDatabase,
      reducers,
      gateway,
      network,
      logger,
    },
  });
  const store = createStore(rootReducer, applyMiddleware(epicMiddleware));
  epicMiddleware.run(rootEpic);
  return store;
};
