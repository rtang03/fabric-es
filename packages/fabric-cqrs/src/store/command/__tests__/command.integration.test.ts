import { getLogger } from '../../../utils';

require('../../../env');
import { enrollAdmin } from '@fabric-es/operator';
import { Wallet, Wallets } from 'fabric-network';
import { pick, values } from 'lodash';
import { Store } from 'redux';
import rimraf from 'rimraf';
import { registerUser } from '../../../account';
import { getNetwork } from '../../../services';
import { PeerOptions } from '../../../types';
import { generateToken } from '../../utils';
import { action } from '../action';
import { getStore } from './__utils__/store';

let context;
let commitId: string;
let store: Store;
const entityName = 'command_test';
const enrollmentId = `command_test${Math.floor(Math.random() * 1000)}`;
const connectionProfile = process.env.CONNECTION_PROFILE;
const channelName = process.env.CHANNEL_NAME;
const fabricNetwork = process.env.NETWORK_LOCATION;
const mspId = process.env.MSPID;
const caUrl = process.env.ORG_CA_URL;
const logger = getLogger({ name: 'command.integration.ts' });

let wallet: Wallet;

beforeAll(async () => {
  try {
    rimraf.sync(`${process.env.WALLET}/${process.env.ORG_ADMIN_ID}.id`);
    rimraf.sync(`${process.env.WALLET}/${process.env.CA_ENROLLMENT_ID_ADMIN}.id`);

    wallet = await Wallets.newFileSystemWallet(process.env.WALLET);

    await enrollAdmin({
      caUrl,
      connectionProfile,
      enrollmentID: process.env.ORG_ADMIN_ID,
      enrollmentSecret: process.env.ORG_ADMIN_SECRET,
      fabricNetwork,
      mspId,
      wallet,
    });

    await enrollAdmin({
      caUrl,
      connectionProfile,
      enrollmentID: process.env.CA_ENROLLMENT_ID_ADMIN,
      enrollmentSecret: process.env.CA_ENROLLMENT_SECRET_ADMIN,
      fabricNetwork,
      mspId,
      wallet,
    });

    await registerUser({
      caAdmin: process.env.CA_ENROLLMENT_ID_ADMIN,
      caAdminPW: process.env.CA_ENROLLMENT_SECRET_ADMIN,
      fabricNetwork,
      enrollmentId,
      enrollmentSecret: 'password',
      connectionProfile,
      wallet,
      mspId,
    });

    context = await getNetwork({
      channelName,
      connectionProfile,
      wallet,
      enrollmentId,
      discovery: true,
      asLocalhost: true,
    });

    context.logger = logger;

    store = getStore(context);
  } catch (e) {
    console.error(e);
    process.exit(1);
  }
});

afterAll(() => {
  rimraf.sync(`${process.env.WALLET}/${enrollmentId}.id`);
  context.gateway.disconnect();
});

describe('CQRS - command Tests', () => {
  // tear-up
  it('should deleteByEntityId', (done) => {
    const tid = generateToken();
    const unsubscribe = store.subscribe(() => {
      const { tx_id, type } = store.getState().write;
      if (tx_id === tid && type === action.DELETE_ERROR) {
        unsubscribe();
        done();
      }
      if (tx_id === tid && type === action.DELETE_SUCCESS) {
        unsubscribe();
        done();
      }
    });
    store.dispatch(
      action.deleteByEntityId({
        tx_id: tid,
        args: { entityName, id: enrollmentId, isPrivateData: false },
        connectionProfile,
        channelName,
        wallet,
      })
    );
  });

  it('should createEntity', (done) => {
    const tid = generateToken();
    const unsubscribe = store.subscribe(() => {
      const { tx_id, result, type } = store.getState().write;
      if (tx_id === tid && type === action.CREATE_SUCCESS) {
        commitId = values(result)[0].commitId;
        expect(pick(values(result)[0], ['entityName', 'version', 'events'])).toMatchSnapshot();
        unsubscribe();
        done();
      }
    });
    store.dispatch(
      action.create({
        tx_id: tid,
        args: {
          entityName,
          id: enrollmentId,
          version: 0,
          events: [{ type: 'User Created', payload: { name: 'me' } }],
          isPrivateData: false,
        },
        // Special attention: createAction will be based on newly created account (given below
        // enrollmentId; to using a new Fabric contract, to submit transaction, and based on its x509
        // cert. Other actions does not require to supply enrollmentId, and will keep using admin ecert
        enrollmentId,
        connectionProfile,
        channelName,
        wallet,
      })
    );
  });

  it('should queryByEntityIdCommitId', (done) => {
    const tid = generateToken();
    const unsubscribe = store.subscribe(() => {
      const { tx_id, result, type } = store.getState().write;
      if (tx_id === tid && type === action.QUERY_SUCCESS) {
        expect(pick(values(result)[0], ['entityName', 'version', 'events'])).toMatchSnapshot();
        unsubscribe();
        done();
      }
    });
    store.dispatch(
      action.queryByEntIdCommitId({
        tx_id: tid,
        args: { entityName, commitId, id: enrollmentId, isPrivateData: false },
        connectionProfile,
        channelName,
        wallet,
      })
    );
  });

  it('should queryByEntityName', (done) => {
    const tid = generateToken();
    const unsubscribe = store.subscribe(() => {
      const { tx_id, result, type } = store.getState().write;
      if (tx_id === tid && type === action.QUERY_SUCCESS) {
        expect(pick(values(result)[0], ['entityName', 'version', 'events'])).toMatchSnapshot();
        unsubscribe();
        done();
      }
    });
    store.dispatch(
      action.queryByEntityName({
        tx_id: tid,
        args: { entityName, isPrivateData: false },
        connectionProfile,
        channelName,
        wallet,
      })
    );
  });

  it('should queryByEntityId', (done) => {
    const tid = generateToken();
    const unsubscribe = store.subscribe(() => {
      const { tx_id, result, type } = store.getState().write;
      if (tx_id === tid && type === action.QUERY_SUCCESS) {
        expect(pick(values(result)[0], ['entityName', 'version', 'events'])).toMatchSnapshot();
        unsubscribe();
        done();
      }
    });
    store.dispatch(
      action.queryByEntityId({
        tx_id: tid,
        args: { entityName, id: enrollmentId, isPrivateData: false },
        connectionProfile,
        channelName,
        wallet,
      })
    );
  });

  it('should deleteByEntityIdCommitId', (done) => {
    const tid = generateToken();
    const unsubscribe = store.subscribe(() => {
      const { tx_id, result, type } = store.getState().write;
      if (tx_id === tid && type === action.DELETE_SUCCESS) {
        expect(result.status).toBe('SUCCESS');
        unsubscribe();
        done();
      }
    });
    store.dispatch(
      action.deleteByEntityIdCommitId({
        tx_id: tid,
        args: { entityName, id: enrollmentId, commitId, isPrivateData: false },
        connectionProfile,
        channelName,
        wallet,
      })
    );
  });
});
