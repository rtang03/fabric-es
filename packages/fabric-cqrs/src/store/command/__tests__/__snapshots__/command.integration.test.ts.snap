// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CQRS - command Tests should createEntity 1`] = `
Object {
  "entityName": "command_test",
  "version": 0,
}
`;

exports[`CQRS - command Tests should queryByEntityId 1`] = `
Object {
  "entityName": "command_test",
  "events": Array [
    Object {
      "payload": Object {
        "name": "me",
      },
      "type": "User Created",
    },
  ],
  "version": 0,
}
`;

exports[`CQRS - command Tests should queryByEntityIdCommitId 1`] = `
Object {
  "entityName": "command_test",
  "events": Array [
    Object {
      "payload": Object {
        "name": "me",
      },
      "type": "User Created",
    },
  ],
  "version": 0,
}
`;

exports[`CQRS - command Tests should queryByEntityName 1`] = `
Object {
  "entityName": "command_test",
  "events": Array [
    Object {
      "payload": Object {
        "name": "me",
      },
      "type": "User Created",
    },
  ],
  "version": 0,
}
`;
